NodeClassifier/utils/reader.py

```diff
@@ -23,36 +22,51 @@ def parse_packet_head(line):
     # Only generate a key if this packet contains IP information
     if len(data) < 2:
         return None
-    if data[2] != 'IP':
-        return None
 
     # Parse out the date and time the packet was seen
     date_str = data[0] + ' ' + data[1]
     date = datetime.datetime.strptime(date_str, '%Y-%m-%d %H:%M:%S.%f')
 
     # Parse out the source and destination addresses and ports
     source_data = data[3].split('.')
-    if len(source_data) < 5:
-        source_port = '0'
-    else:
-        source_port = source_data[4]
-
-    source_str = '.'.join(source_data[0:4]) + ':' + source_port
-
     destination_data = data[5].split('.')
-    if len(destination_data) < 5:
-        destination_port = '0'
-        destination_str = '.'.join(destination_data[0:4])[0:-1] \
-                          + ':' \
-                          + destination_port
-
+    destination_port = '0'
+    source_port = '0'
+
+    # ipv4 packet
+    if data[2] == 'IP':
+        # if TCP or UDP set port
+        if len(source_data) >= 5:
+            source_port = source_data[4]
+
+        source_str = '.'.join(source_data[0:4]) + ':' + source_port
+        if len(destination_data) < 5:
+            destination_str = '.'.join(destination_data[0:4])[0:-1] \
+                              + ':' \
+                              + destination_port
+        else:
+            destination_port = destination_data[4][0:-1]
+            destination_str = '.'.join(destination_data[0:4]) \
+                              + ':' \
+                              + destination_port
+    # ipv6 packet
+    elif data[2] == 'IP6':
+        # if TCP or UDP set port
+        if len(source_data) >= 2:
+            source_port = source_data[1]
+
+        source_str = source_data[0] + ':' + source_port
```

Comment:
Could you do me a favor and change the SIP6/DIP6-ports separators from ':' to '.' ? Otherwise, this section looks good.

Reply:
If we do that, then how we can separate IP and port in featurizer.
which use `:` delimiter to extract  IP and port from the key?

Reply:
You know... good point. 🤔